### PR TITLE
Tech: improve treeshaking - add dist stats file (for comparison)

### PR DESCRIPTION
**Pull request by ndelangen**
- Update the bundle script to do more tree-shaking
- Don't generate `.mjs`-files for code that runs exclusively in node (sometime in 8.0 we'll revisit this when we go ESM-only)
- Update the `dist-size` stats file, to show effects of the bundler change

### CODE REVIEW

File: scripts/prepare/bundle.ts [hunk 1]
```diff
@@ -75,26 +102,25 @@ const run = async ({ cwd, flags }: { cwd: string; flags: string[] }) => {
...
+        env: {
+          NODE_ENV: 'production',
+        },
         format: ['esm'],
         target: 'chrome100',
         clean: !watch,
         ...(dtsBuild === 'esm' ? dtsConfig : {}),
         platform: platform || 'browser',
-        esbuildPlugins: [
-          aliasPlugin({
```
Comment (valentinpalkovic) [inline scripts/prepare/bundle.ts:88]: Is this something which is already built into rollup and we don't need this anymore?

Comment (ndelangen) [reply scripts/prepare/bundle.ts:88]: I'm not sure if this means:

> tsup offers an additional option to let you use Rollup for tree shaking instead:

esbuild won't be used at all. Or if rollup will be used for some part on top of esbuild...

Comment (valentinpalkovic) [reply scripts/prepare/bundle.ts:88]: As I understand the code, rollup is used on top of esbuild as a kind of esbuild plugin. But let's be 100% sure :)

Comment (ndelangen) [reply scripts/prepare/bundle.ts:88]: Yes, got confirmation that the above information is correct

**Pull request merged**